client/network/network.js

```diff
@@ -30,19 +27,22 @@ let ready, readyPromise = new Promise(r => ready = r);
 export function init() {
   socket = io(location.host);
   socket.on('reconnect', () => {
+    // When we reconnect after a disconnection, we need to tell the server
+    // about who we are all over again.
     socket.emit('client-start', {rect: info.virtualRectNoBezel.serialize()});
   });
+  // Install our time listener.
   socket.on('time', time.adjustTimeByReference);
+  // Install the machinery for our per-module network.
+  installMessageHandler(socket);
 
+  // Tell the server who we are.
   socket.emit('client-start', {rect: info.virtualRectNoBezel.serialize()});
```

Comment:
Abstract this? Weird to have two copies of something so critically specific.

Reply:
Good point.